docs/tutorials/rest_api/index.md

```diff
@@ -0,0 +1,412 @@
+
+2. **Feldera platform:** This tutorial requires you to have a running Feldera platform.
+   If you do not have one already, you can start one locally using
+   [**docker**](https://docs.docker.com/engine/install/):
+   ```
+   curl https://raw.githubusercontent.com/feldera/feldera/main/deploy/docker-compose.yml | \
+   docker compose -f - up
+   ```
+   (leave it running in a separate terminal while going through this tutorial)
+
+   > For the remainder of this tutorial, we will use **http://localhost:8080** as
+   > this is the customary local hostname:port for test setups. You will need change
+   > it to match the Feldera platform you are using.
+
+3. **(Optional) API key:** Skip this step if you are using a local test setup
+   using Docker as described above. If the Feldera platform requires authentication,
+   you must generate an API key in the Web Console at the _Settings_ tab.
+   You can add it to a `curl` call in the following way:
+   ```
+   curl -i -H "Authorization: <API-KEY>" -X GET http://localhost:8080/v0/programs
+   ```
+
+   > For the remainder of this tutorial, you will need to add
+   > `-H "Authorization: <API-KEY>"` to each of the calls.
+
+4. **Check it's working:** You can verify it's working by running:
+   ```
+   curl -i -X GET http://localhost:8080/v0/programs
+   ```
+   ... this should output `HTTP/1.1 200 OK` followed by some HTTP headers
+   and finally below it the result: an JSON array of program objects, which
+   when there are none (yet!) would be `[]`.
+
+**Congratulations, you've already done your first direct API interaction!**
+
+## Getting started
+
+In this tutorial, we are going to approach a use case of supply chain management.
+It focuses on identifying the vendors with the lowest prices for parts.
+We will create a pipeline which ingests data from several HTTP sources,
+perform several interesting queries on them, and read from the incremental views.
+
+We'll be going through the following steps:
+
+1. Create and compile an SQL program
+2. Create data connectors
+3. Create a pipeline with the prepared program and connectors
+4. Start the pipeline
+5. Check pipeline progress
+6. Read data directly from an incremental view
+7. Feed data directly into a stream table
+8. Cleanup
+
+... all using just `curl`!
+
+> Note: at any point in the tutorial, don't forget you can check out the
+Web Console by visiting `http://localhost:8080` in your browser!
+
+### Step 1: SQL program
+
+The SQL program defines the input stream tables using `CREATE TABLE` statements
+and incremental views using `CREATE VIEW` statements. We will create three tables,
+namely `vendor`, `part`, and `price`. As views, we'll create `low_price` which lists
+the lowest available price for each part across all vendors, and `preferred_vendor` which
+supplements the lowest price by adding vendor information.
+
+We create the program named `sc-program` as follows:
+
+```
+curl -i -X PUT http://localhost:8080/v0/programs/sc-program \
```

Comment:
Let's avoid this much use of escape characters. Either use single quotes. Or create a file with the body and then use `--data @path-to-file`.

Reply:
Single quotes unfortunately do not work as then the newlines characters become literal, and the JSON parser does not accept multiline strings. I changed it to input a data file as you suggested, with the minimum amount of newlines I could manage and nicest layout. I don't know how to improve it further, any thoughts are very welcome :)

Reply:
Can be improved in a next iteration